nav2_costmap_2d/plugins/costmap_filters/keepout_filter.cpp

```diff
@@ -148,57 +156,80 @@ void KeepoutFilter::process(
     return;
   }
 
-  double wx, wy;  // world coordinates
-
-  // Optimization: iterate only in overlapped
-  // (min_i, min_j)..(max_i, max_j) & mask_costmap_ area.
-  //
-  //           mask_costmap_
-  //       *----------------------------*
-  //       |                            |
-  //       |                            |
-  //       |      (2)                   |
-  // *-----+-------*                    |
-  // |     |///////|<- overlapped area  |
-  // |     |///////|   to iterate in    |
-  // |     *-------+--------------------*
-  // |    (1)      |
-  // |             |
-  // *-------------*
-  //  master_grid (min_i, min_j)..(max_i, max_j) window
-  //
-  // ToDo: after costmap rotation will be added, this should be re-worked.
-
-  // Calculating bounds corresponding to bottom-left overlapping (1) corner
-  int mg_min_x, mg_min_y;  // masger_grid indexes of bottom-left (1) corner
-  // mask_costmap_ -> master_grid intexes conversion
-  const double half_cell_size = 0.5 * mask_costmap_->getResolution();
-  wx = mask_costmap_->getOriginX() + half_cell_size;
-  wy = mask_costmap_->getOriginY() + half_cell_size;
-  master_grid.worldToMapNoBounds(wx, wy, mg_min_x, mg_min_y);
-  // Calculation of (1) corner bounds
-  if (mg_min_x >= max_i || mg_min_y >= max_j) {
-    // There is no overlapping. Do nothing.
-    return;
-  }
-  mg_min_x = std::max(min_i, mg_min_x);
-  mg_min_y = std::max(min_j, mg_min_y);
-
-  // Calculating bounds corresponding to top-right window (2) corner
-  int mg_max_x, mg_max_y;  // masger_grid indexes of top-right (2) corner
-  // mask_costmap_ -> master_grid intexes conversion
-  wx = mask_costmap_->getOriginX() +
-    mask_costmap_->getSizeInCellsX() * mask_costmap_->getResolution() + half_cell_size;
-  wy = mask_costmap_->getOriginY() +
-    mask_costmap_->getSizeInCellsY() * mask_costmap_->getResolution() + half_cell_size;
-  master_grid.worldToMapNoBounds(wx, wy, mg_max_x, mg_max_y);
-  // Calculation of (2) corner bounds
-  if (mg_max_x <= min_i || mg_max_y <= min_j) {
-    // There is no overlapping. Do nothing.
-    return;
+  tf2::Transform tf2_transform;
```

Comment:
Set this to identity, just in case

Reply:
Not sure, what did you mean in this comment?

Reply:
Yes, exactly